### PR TITLE
Fix how `appearanceTools` works

### LINKED ISSUE #37232: In 12.1, blockGap appears partially enabled when it shouldn't be
### Description

In 12.1, something changed (still trying to locate) that enables `blockGap` related CSS even if `blockGap` is not enabled in theme.json. I can also confirm that this issue exists in 5.9 Beta 2 without the Gutenberg plugin active.

In Gutenberg 12.0.2, I have a Group block with a Paragraph and Buttons block inside. It looks something like this:
image

Now in 12.1, the same collection of blocks looks like this. 
image

Notice all the additional CSS applied to the paragraph block. In theme.json, `blockGap` is not enabled on this theme.

### Step-by-step reproduction instructions

1. Install Twenty Twenty-Two and disable `blockGap` in theme.json
2. Install Gutenberg 12.0.2
3. Create a Group block and place a few paragraph blocks inside. 
4. View the markup on the frontend. 
5. Upgrade Gutenberg to 12.1 
6. Review the markup again and see all the additional `blockGap` related CSS added to the frontend. 

### Screenshots, screen recording, code snippet

_No response_

### Environment info

WordPress 5.9 Beta 2, Gutenberg 12.1

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

### PR DESCRIPTION
**Pull request by oandregal**
Fixes https://github.com/WordPress/gutenberg/issues/37232
Follow-up to https://github.com/WordPress/gutenberg/pull/36646

This PR fixes two things:

- It looks like the mere presence of the `appearanceTools` flag was activating the others, while it should only be the case if `appearanceTools` was `true`.
- There are some flags (such as `blockGap`) for which the `null` value is actually valid, so we need to detect whether they are set/unset using a different marker.

## How to test

- Activate TwentyTwentyTwo theme.
- Add a post with a group and a few paragraphs within. Publish.

Verify that without `appearanceTools` things work as expected:

- Remove or set to `false` the flag `appearanceTools` in the `theme.json` of TwentyTwentyTwo.
- Load the published post in the front-end and verify that the block gap classes are not present (see https://github.com/WordPress/gutenberg/issues/37232 for details).

Verity that with `appearanceTools` active blockGap can be disabled:

- Set to `true` the flag `appearanceTools` in the `theme.json` of TwentyTwentyTwo.
- Set `settings.spacing.blockGap` to `null`.
- Load the published post in the front-end and verify that the block gap classes are not present (see https://github.com/WordPress/gutenberg/issues/37232 for details).

### CODE REVIEW

**Comment by oandregal** [comment]
@ndiego ~This won't fix the issue you reported at https://github.com/WordPress/gutenberg/issues/37232 because it's different. I'll comment there. This fix is still necessary, though.~

Let me rephrase: this PR fixes a couple of issues with `appearanceTools` that affected the `blockGap`. What I mean is that `blockGap: false` doesn't work as expected and needs to be investigated separately.

**Comment by ndiego** [comment]
Thanks for this @oandregal, I have tested and confirm that this fix works as intended!

**Review by ndiego — APPROVED** [review]

**Review by andrewserong — APPROVED** [review]
Thanks for the ping and fix @oandregal!

✅ Tested that the `appearanceTools` setting only affects the opt-ins when set to `true`.
✅ Tested that `null` is a valid override value in `settings.spacing.blockGap` and that the settings overrides appear to work correctly with the `appearanceTools` setting.

LGTM! 🚀

**Review by aaronrobertshaw — APPROVED** [review]
Nice work as always @oandregal  👍 

✅ code changes LGTM
✅ unit tests pass
✅ `appearanceTools` only opts-in other features when set to `true`
✅ `null` override for `settings.spacing.blockGap` works
